src/ODESolvers/MultirateInfinitesimalStepMethod.jl

```diff
@@ -0,0 +1,276 @@
+                           α, β, γ,
+                           Q::AT; dt=0, t0=0) where {AT<:AbstractArray}
+
+This is a time stepping object for explicitly time stepping the partitioned differential
+equation given by right-hand-side functions `f_fast` and `f_slow` with the state `Q`, i.e.,
+
+```math
+  \\dot{Q} = f_fast(Q, t) + f_slow(Q, t)
+```
+
+with the required time step size `dt` and optional initial time `t0`.  This
+time stepping object is intended to be passed to the `solve!` command.
+
+The constructor builds a multirate infinitesimal step Runge-Kutta scheme
+based on the provided `α`, `β` and `γ` tableaux and `fastmethod` for solving
+the fast modes.
+
+### References
+    @article{KnothWensch2014,
+      title={Generalized split-explicit Runge--Kutta methods for the compressible Euler equations},
+      author={Knoth, Oswald and Wensch, Joerg},
+      journal={Monthly Weather Review},
+      volume={142},
+      number={5},
+      pages={2067--2081},
+      year={2014}
+    }
+"""
+struct MultirateInfinitesimalStep{T, RT, AT, FS, Nstages, Nstagesm1, Nstagesm2, Nstages_sq} <: ODEs.AbstractODESolver
+  "time step"
+  dt::Array{RT, 1}
+  "time"
+  t::Array{RT, 1}
+  "storage for y_n"
+  yn::AT
+  "Storage for ``Y_nj - y_n``"
+  ΔYnj::NTuple{Nstagesm2, AT}
+  "Storage for ``f(Y_nj)``"
+  fYnj::NTuple{Nstagesm1, AT}
+  "Storage for offset"
+  offset::AT
+  "slow rhs function"
+  slowrhs!
+  "RHS for fast solver"
+  tsfastrhs!::TimeScaledRHS{RT}
+  "fast rhs method"
+  fastsolver::FS
+  "number of substeps per stage"
+  nsubsteps::Int
+  α::SArray{NTuple{2, Nstages}, RT, 2, Nstages_sq} 
+  β::SArray{NTuple{2, Nstages}, RT, 2, Nstages_sq}
+  γ::SArray{NTuple{2, Nstages}, RT, 2, Nstages_sq}
+  d::SArray{NTuple{1, Nstages}, RT, 1, Nstages} 
+  c::SArray{NTuple{1, Nstages}, RT, 1, Nstages} 
+  c̃::SArray{NTuple{1, Nstages}, RT, 1, Nstages} 
+  function MultirateInfinitesimalStep(slowrhs!, fastrhs!, fastmethod, nsubsteps,
+                                      α, β, γ,
+                                      Q::AT; dt=0, t0=0) where {AT<:AbstractArray}
+
+    T = eltype(Q)
+    RT = real(T)
+    dt = [dt]
+    t0 = [t0]
+
+    Nstages = size(α, 1)
+
+    yn = similar(Q)
+    ΔYnj = ntuple(_ -> similar(Q), Nstages-2)
+    fYnj = ntuple(_ -> similar(Q), Nstages-1)
+    offset = similar(Q)
+    tsfastrhs! = TimeScaledRHS(RT(0), RT(0), fastrhs!)
+    fastsolver = fastmethod(tsfastrhs!, Q)
+
+    d = sum(β, dims=2)
+
+    c = similar(d)
+    for i = eachindex(c)
+      c[i] = d[i]
+      if i > 1
+        c[i] += sum(j-> (α[i,j] + γ[i,j])*c[j], 1:i-1)
+      end
+    end
+    c̃ = α*c
+    
+    new{T, RT, AT, typeof(fastsolver), Nstages, Nstages-1, Nstages-2, Nstages ^ 2}(dt, t0, yn, ΔYnj, fYnj, offset,
+                                         slowrhs!, tsfastrhs!, fastsolver, nsubsteps,
+                                         α, β, γ, d, c, c̃)
+  end
+end
+
+
+# TODO almost identical functions seem to be defined for every ode solver,
```

Comment:
I'm fine with that. Had noticed the same thing once I started adding the initial multirate stuff.